### DISCUSSION TITLE
对Web3URL的新理解，以及是不是可以结合ZK？

### DISCUSSION THREAD

**Discussion by zedz** [IntensiveCoLearning/Web3-URL#130] [Q&A] [answered]
来自微信好友，感谢他的帮助！但他没学，**我需要有人帮我确认以下这个理解对不对就好**：

1. 目前的web2网站都是一个**前端**一个**后端**，**前端**和**后端**都需要放在**中心化**的服务器上。
2. 目前的web3的dApp，也是把**前端**放在**中心化服务器**上去调用**去中心化**的**智能合约（后端）**
3. 有了web3url和ethStorage，可以把前端也放在去中心化的链上，通过web3url调用
4. 从而实现全链去中心化（域名ens，前端html/js/css，存储）
5. 所以web3url要**同时**调用一个**智能合约**和一个**链上存储好的前端文件**，来显示dApp或者网站。

BTW结合这两天新看的ZK的一些知识，感觉web3url可不可以调用在链下的前端和后端文件呢？通过ZK Validium来证明一下数据有效性？这样可能可以用更低的费用做到去中心化？

**Comment by qizhou** [comment, marked as answer]
> 来自微信好友，感谢他的帮助！但他没学，**我需要有人帮我确认以下这个理解对不对就好**：
> 
> 1. 目前的web2网站都是一个**前端**一个**后端**，**前端**和**后端**都需要放在**中心化**的服务器上。
> 2. 目前的web3的dApp，也是把**前端**放在**中心化服务器**上去调用**去中心化**的**智能合约（后端）**
> 3. 有了web3url和ethStorage，可以把前端也放在去中心化的链上，通过web3url调用
> 4. 从而实现全链去中心化（域名ens，前端html/js/css，存储）
> 5. 所以web3url要**同时**调用一个**智能合约**和一个**链上存储好的前端文件**，来显示dApp或者网站。
> 
非常好的总结！

> BTW结合这两天新看的ZK的一些知识，感觉web3url可不可以调用在链下的前端和后端文件呢？通过ZK Validium来证明一下数据有效性？这样可能可以用更低的费用做到去中心化？

EthStorage在解决的这个问题。EthStorage是一个以太坊的存储layer 2：
- 链下保存需要存储的前端/后端的数据（文件）
- 链上保存对应数据的哈希（为更好兼容以太坊，使用EIP-4844 BLOB的格式）

最后，通过我们的ZK 存储证明来去通过链上合约来验证链下数据的冗余度，作为存储激励分配的指标。

具体细节可以参见官网 ethstorage.io

**Reply by zedz** [reply]
谢谢回复！今天学习了ethStorage 好像刚好就是上面理解的样子！似乎L2就是链下的意思了？

1. 用户通过一个应用来调用应用的合约，上传数据到DAN，这个合约同时会去和ethStorage合约在L1进行交互。
2. 和ethStorage交互的时候，ethStorage会广播给L2的Storage Providers
3. Storage Provider们从DAN下载数据
4. Storage Provider们把存储证明提交到L1，证明在L2中有大量的用户上传的数据副本
5. ethStorage合约奖励这些成功提供有效存储证明的Storage Provider们

---

看图好像ethStorage可以和各种rollups加起来用，是不是用不同的rollups就不一定是用ZK的存储证明？就是用其他的rollups技术？和其他的rollups一起用相当于到了L3吗？如果跟ZK rollups一起用，是可以兼具最高的TPS和最高的data吗？

**Reply by linghuccc** [reply]
> > 来自微信好友，感谢他的帮助！但他没学，**我需要有人帮我确认以下这个理解对不对就好**：
> > 
> > 1. 目前的web2网站都是一个**前端**一个**后端**，**前端**和**后端**都需要放在**中心化**的服务器上。
> > 2. 目前的web3的dApp，也是把**前端**放在**中心化服务器**上去调用**去中心化**的**智能合约（后端）**
> > 3. 有了web3url和ethStorage，可以把前端也放在去中心化的链上，通过web3url调用
> > 4. 从而实现全链去中心化（域名ens，前端html/js/css，存储）
> > 5. 所以web3url要**同时**调用一个**智能合约**和一个**链上存储好的前端文件**，来显示dApp或者网站。
> 
> 非常好的总结！
> 
> > BTW结合这两天新看的ZK的一些知识，感觉web3url可不可以调用在链下的前端和后端文件呢？通过ZK Validium来证明一下数据有效性？这样可能可以用更低的费用做到去中心化？
> 
> EthStorage在解决的这个问题。EthStorage是一个以太坊的存储layer 2：
> 
> * 链下保存需要存储的前端/后端的数据（文件）
> * 链上保存对应数据的哈希（为更好兼容以太坊，使用EIP-4844 BLOB的格式）
> 
> 最后，通过我们的ZK 存储证明来去通过链上合约来验证链下数据的冗余度，作为存储激励分配的指标。
> 
> 具体细节可以参见官网 ethstorage.io

“链下保存需要存储的前端/后端的数据（文件）”
这个“链下”具体指的是哪里？

如果数据实际存储是在链下，链上只存哈希，那和 IPFS 有什么本质区别？

**Reply by qzhodl** [reply]
> > > 来自微信好友，感谢他的帮助！但他没学，**我需要有人帮我确认以下这个理解对不对就好**：
> > > 
> > > 1. 目前的web2网站都是一个**前端**一个**后端**，**前端**和**后端**都需要放在**中心化**的服务器上。
> > > 2. 目前的web3的dApp，也是把**前端**放在**中心化服务器**上去调用**去中心化**的**智能合约（后端）**
> > > 3. 有了web3url和ethStorage，可以把前端也放在去中心化的链上，通过web3url调用
> > > 4. 从而实现全链去中心化（域名ens，前端html/js/css，存储）
> > > 5. 所以web3url要**同时**调用一个**智能合约**和一个**链上存储好的前端文件**，来显示dApp或者网站。
> > 
> > 
> > 非常好的总结！
> > > BTW结合这两天新看的ZK的一些知识，感觉web3url可不可以调用在链下的前端和后端文件呢？通过ZK Validium来证明一下数据有效性？这样可能可以用更低的费用做到去中心化？
> > 
> > 
> > EthStorage在解决的这个问题。EthStorage是一个以太坊的存储layer 2：
> > 
> > * 链下保存需要存储的前端/后端的数据（文件）
> > * 链上保存对应数据的哈希（为更好兼容以太坊，使用EIP-4844 BLOB的格式）
> > 
> > 最后，通过我们的ZK 存储证明来去通过链上合约来验证链下数据的冗余度，作为存储激励分配的指标。
> > 具体细节可以参见官网 ethstorage.io
> 
> “链下保存需要存储的前端/后端的数据（文件）” 这个“链下”具体指的是哪里？
> 
> 如果数据实际存储是在链下，链上只存哈希，那和 IPFS 有什么本质区别？

链下的存储服务商需要向链上提供存储证明，由智能合约来验证确实存储了数据才会发送存储奖励，整个过程的安全性是有L1来保证安全的。

**Reply by zedz** [reply]
所以同问：这个“链下”具体指的是哪里？本地磁盘也可以吗？是不是只要可以提供储存证明就可以？

**Reply by qzhodl** [reply]
链下指的是EthStorage的layer 2的存储服务提供商，他们保存在自己的磁盘中，并向链上提供存储证明
